From d6e863878c27bc3d186187df4dc9d0aea7cd1c9d Mon Sep 17 00:00:00 2001
From: Thibaud Michaud <thibaudm@chromium.org>
Date: Thu, 09 Jan 2025 17:02:00 +0100
Subject: [PATCH] Merged: [wasm] Add missing entry to wasm builtin list

Add the invalid wasm-to-js signature wrapper to the list of wasm
builtins. This builtin is called from wasm and calls a runtime function,
so it needs to switch to the central stack if needed.

R=clemensb@chromium.org

Fixed: 383854201
(cherry picked from commit 0cb861fc179100f9d4c12512ec9cf1aa45d0e17f)

Change-Id: I246d0681c35f242f2875e8824ec9b13ae49db773
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/6163547
Reviewed-by: Clemens Backes <clemensb@chromium.org>
Commit-Queue: Thibaud Michaud <thibaudm@chromium.org>
Cr-Commit-Position: refs/branch-heads/13.2@{#56}
Cr-Branched-From: 24068c59cedad9ee976ddc05431f5f497b1ebd71-refs/heads/13.2.152@{#1}
Cr-Branched-From: 6054ba94db0969220be4f94dc1677fc4696bdc4f-refs/heads/main@{#97085}
---

--- a/v8/src/wasm/wasm-builtin-list.h
+++ b/v8/src/wasm/wasm-builtin-list.h
@@ -146,6 +146,7 @@
   V(DeoptimizationEntry_Eager)                                                 \
   V(WasmLiftoffDeoptFinish)                                                    \
   V(WasmPropagateException)                                                    \
+  V(WasmToJsWrapperInvalidSig)                                                 \
   V(WasmTrapHandlerThrowTrap)                                                  \
   IF_SHADOW_STACK(V, AdaptShadowStackForDeopt)
 
